### ISSUE TITLE
Discord AppIndicator freezes GNOME for a second when using voice input

### ISSUE THREAD

**Issue by glossed** [ubuntu/gnome-shell-extension-appindicator#171]
### Description
Using voice input on Discord with this extension enabled causes GNOME shell to freeze for a split second. For example, talking in a group call or voice chat while watching a YouTube video will cause only the video to freeze for a split second everytime Discord registers your microphone input.
### Environment

- OS: Arch Linux (tested and occurs on Ubuntu and Manjaro)
- GNOME: 3.32
- Extension version: Latest from `extensions.gnome.org`
### Reproduction steps
- Install extension from `extensions.gnome.org` and enable it
- Install Discord however you like whether it be Flatpak, Snap, AUR, etc.
- Run Discord and join a voice chat or voice call
- Speak while watching a video or opening menus, etc.
- Actions will freeze for a split second when your microphone is registered
### Hardware
- Machine: HP Z210 Workstation
- CPU: Intel Core i7-2600
- RAM: 16GB DDR3
- GPU: NVIDIA GeForce GTX 1050 Ti on driver 418.56

**Comment by blastermaster77** [comment]
I can confirm it happens to me on Ubuntu 19.04 with discord, the icon starts to flash some times when it detects microphone input and the desktop stutters and video too. I installed the topiconsplus inplementation and it does not have this problem, hope there is a fix soon. https://www.youtube.com/watch?v=ABavT7yXlso I sent this video to de developers in request of them for the problem.

**Comment by TheGreatestJannet** [comment]
I am also experiencing this on Fedora 30. Is there a solution?

**Comment by vanvugt** [comment]
Duplicate of #157 ?

**Comment by rowbawts** [comment]
Same issue on Fedora 31, still nothing?

**Comment by TheGreatestJannet** [comment]
It seems to be a bug with Discord and not this extension. Seeing as Discord has zero Linux developers and they don’t actually care about their Linux program don’t expect a fix for this any time soon...

**Comment by linusfolke** [comment]
Well discord probably does something wrong and wont be fixed, however this problem does not appear when having the indicator in example a kde session, so perhaps there is something that can be done to the indicator?

**Comment by AktasC** [comment]
We need a fix for this, don't wanna change my DE just to please Discord's devs.

**Comment by scj643** [comment]
The issue seems to also involve a memory leak in gnome shell. 

Before connecting to voice
![image](https://user-images.githubusercontent.com/5656510/74351187-c9824100-4d84-11ea-857a-3cdbabf4f79d.png)

After connecting to voice and speaking.
![image](https://user-images.githubusercontent.com/5656510/74351322-fa627600-4d84-11ea-9d78-789e14e820fc.png)

**Comment by scj643** [comment]
I made a fork that excludes Discord from app icon updates here https://github.com/scj643/gnome-shell-extension-appindicator/commit/99e366f0e57c2383841e4dc714d4e71cf9127ebe

**Comment by scj643** [comment]
This issue seems to affect other Electron based apps as well. One example is Google Play Music Desktop.

**Comment by scj643** [comment]
So anyone saying this is a Discord issue should rethink that. This is an extension issue due to the fact that other electron apps have this issue as well.

**Comment by mei-li** [comment]
It happens to me with Slack as well. I see this in my syslog
`gnome-shell[2667]: [AppIndicatorSupport-FATAL] unable to lookup icon for`
and the system freezes completely and I have to hard reboot. 
I don't have a clear way to reproduce it yet, sometimes I am using the microphone when it happens and sometimes not.

**Comment by 3v1n0** [comment]
Please, try latest master and let me know if this has been addressed.

**Comment by chr-debhl** [comment]
No freezes here while running `master` (neither in shell or videos, as mentioned above). However, cursor movement is laggy. journalctl is spammed with [this error](https://i.imgur.com/UEUpV0B.png).

Running GNOME 3.36 on Wayland.

**Comment by 3v1n0** [comment]
> No freezes here while running `master` (neither in shell or videos, as mentioned above). However, cursor movement is laggy. journalctl is spammed with [this error](https://i.imgur.com/UEUpV0B.png).
> 
> Running GNOME 3.36 on Wayland, latest Discord from flathub.

Ok that erro should be now handled by commit c5b31a64da15e1f1359083fdebda7ad3bf8f82f3, so I think we can close this issue.

Fixed by commit 8b8455906f2b126031e5899e2b3e7ce6c247cab7

**Comment by linusfolke** [comment]
Just tried master, no issues while running a game and speaking in discord, however its still noticable for me in shell and when watching videos while speaking in discord.

**Comment by 3v1n0** [comment]
The icon is animating while doing that, right?

In such case, we can't really do much... Unless fix the shell itself I think.

**Comment by linusfolke** [comment]
There is no animation, however the icon remains now, previous the icon would disappear for a split second when starting and stopping to talk in discord. The freeze now isn't as bad as before but there still is one.

**Comment by fenrig** [comment]
I'm playing wow on an RTX 2080 and when I have the dedicated discord app and this extension, it still freezes. So can confirm it's not fixed

**Comment by fenrig** [comment]
The freezes were gone when using this fork/branch in gnome 3.34:
https://github.com/scj643/gnome-shell-extension-appindicator/commit/999b58d6d889ecd6b7b641ed4ba71c7cf1d4b8b1

**Comment by fenrig** [comment]
So for everyone I reapplied the patch of that fork and freezes are gone again <3

**Comment by sondr3** [comment]
Same for me, the latest release from `master` didn't fix this but the patch from @scj643 makes my desktop usable again while using Discord.

**Comment by agrisvv** [comment]
fedora 31. I can confirm what mouse lagging on wayland gone only after blacklisted icons like sondr3. Issue started today after run updates for system.
App are Mattermost - electron.
Not by this topic but if installed from flathub, then icons only shows after disable/enable extension after mattermost is started

**Comment by crscillitoe** [comment]
ubuntu 20 here, same issue.

I would prefer if my DE could take priority over app icon animations.

**Comment by jbeaudoin11** [comment]
Same here, Zorin OS 15.2 (ubuntu 18.04)
discord 0.0.10 (tried both snap and .deb)
I was able to manually edit the appindicator and apply the patch of @scj643

Something is really weird tho.
This problem appeared when i moved my OS partition to an other drive and pc.
PC 1 (personal) :
CPU : AMD r7 3700x
GPU : Nvidia 2080 ti (i don't rember the driver version, pretty sure it was 4xx)
Motherboard : MSI MPG X570 GAMING EDGE WIFI
OS Drive : Sata SSD

PC 2 (work) :
CPU : AMD r7 3700x
GPU : Nvidia 1070 (tried w/ drivers 410, 435, 440, 450)
Motherboard : Asus TUF GAMING X570-PLUS (WI-FI)
OS Drive : NVMe 970 evo plus 

Everything worked fine on my personal pc..
Discord was working and no lag from gnome-shell
I also update my os every morning, i would be suprise if an update did that.
The only thing i can see is the nvidia driver or the gpu gen model (2000 vs 1000 series).
But @fenrig says he tried with a 2080 so :man_shrugging:

**Comment by AktasC** [comment]
@jbeaudoin11 Got the same issue here with an R7 3700X & RX 5600XT .
I don't think the problem's your GPU, I recently changed from a 1070 to a 5600XT and still experience it.
My first idea, after reading comments here, was that XServer might be the culprit but since people also experience this on Wayland I have no clue as to what causes this mess..

**Comment by scj643** [comment]
I've given up on app indicators recently though I might try and fork this with a way to log how often an app changes it's icon and add a UI to allow adding apps to the icon blacklist.

**Comment by ryannathans** [comment]
Pop_OS! 

This affects everyone I know - I am not even using app indicators and the discord app doesn't change icon

**Comment by ryannathans** [comment]
The launchpad bug was closed despite not really fixing the problem https://bugs.launchpad.net/ubuntu/+source/gnome-shell-extension-appindicator/+bug/1849142

**Comment by amithiel** [comment]
Hi, can i ask why is this closed? i'm with ubuntu 20.04.1 and latest gnome-shell-indicator and its clearly not fixed

**Comment by scj643** [comment]
@3v1n0 This issue is still not fixed. On Fedora 33 enabling the extension and using Discord still causes a memory leak.

**Comment by onelegchair** [comment]
Still having this issue on Pop_OS 20.10

**Comment by Kneemund** [comment]
I'm also having the memory leak issue that scj643 pointed out (Arch BTW). I guess I'll try using their fork instead. No idea why this is closed.

**Comment by arshetamine** [comment]
Still not fixed, the only fix available is to add discord to blacklist in appIndicator.js (ubuntu 20.10)

**Comment by fbnlsr** [comment]
Chiming in to confirm the problem still exists on Pop!_OS 20.10. Whenever my USB mic turns on or off, the whole OS freezes for a split second.

**Comment by jacksongoode** [comment]
Bump! Keep this alive!

**Comment by arsovskidev** [comment]
Update: viber started doing the same thing when receiving message as same as talking on discord.

**Comment by jacksongoode** [comment]
Just wanted to follow up, for those on Ubuntu 20.04, disable the system AppIndicator extension and reinstall the latest version from the Extensions page.

**Comment by fbnlsr** [comment]
Still having this issue on Pop!_OS 20.10. Anyone updated to 21.04 yet?

**Comment by onelegchair** [comment]
I'm on Pop 21.04 and still having the issue.

**Comment by s-bose** [comment]
Pop os 21.04. Still having this issue. Tried both .deb and flatpak.

**Comment by arsovskidev** [comment]
> Pop os 21.04. Still having this issue. Tried both .deb and flatpak.

Still i was not able to fix this... stopped using discord and viber 🙁

**Comment by fbnlsr** [comment]
Follow up: I'm now forced to use the web client whenever I'm in the need of an audio/video chat. No freezes there. If I use audio/video chat in the Electron client, the OS freezes every time the mic opens up/closes.

**Comment by s-bose** [comment]
For me, I was able to fix the issue by changing the icon theme. For some reason, the papirus icon pack was the issue for me, so if you are using papirus perhaps try switching to a different icon pack and see if it makes any difference.

**Comment by arsovskidev** [comment]
> For me, I was able to fix the issue by changing the icon theme. For some reason, the papirus icon pack was the issue for me, so if you are using papirus perhaps try switching to a different icon pack and see if it makes any difference.

Will try this asap because i am using the papirus too! Thanks

**Comment by arsovskidev** [comment]
The freezes can be reproduced by running applications like viber,discord,slack on another workspace or monitors.
